packages/gafl-webapp-service/src/pages/summary/contact-summary/__tests__/route.spec.js

```diff
@@ -64,318 +101,190 @@ const generateRequestMock = (permission, query, status = {}) => ({
+      async urlToCheck => {
+        const sampleRequest = getRequestMock(getMockPermission({}))
+        await getData(sampleRequest)
+        expect(addLanguageCodeToUri).toHaveBeenCalledWith(sampleRequest, urlToCheck)
+      }
+    )
+  })
 
-      expect(changeLicenceDetails).toMatchSnapshot()
+  describe('checkNavigation', () => {
+    it('should throw a GetDataRedirect if licence-fulfilment page is false on the status', async () => {
+      const status = {
+        [ADDRESS_ENTRY.page]: false,
+        [ADDRESS_SELECT.page]: false
+      }
+      const mockRequest = getRequestMock({ permission: {}, status })
+      await expect(() => getData(mockRequest)).rejects.toThrowRedirectTo(ADDRESS_LOOKUP.uri)
     })
 
-    describe('addLanguageCodeToUri', () => {
-      beforeEach(jest.clearAllMocks)
-
-      it.each([[ADDRESS_LOOKUP.uri], [LICENCE_FULFILMENT.uri], [LICENCE_CONFIRMATION_METHOD.uri], [CONTACT.uri], [NEWSLETTER.uri]])(
-        'test addLanguageCodeToUri is called correctly',
-        async urlToCheck => {
-          const permission = {
-            permit: {
-              cost: 1
-            },
-            licenceLength: '12M',
-            isLicenceForYou: true,
-            licensee: {
-              birthDate: '1996-01-01',
-              postalFulfilment: true
-            }
-          }
-
-          const sampleRequest = generateRequestMock(permission)
-
-          await getData(sampleRequest)
-
-          expect(addLanguageCodeToUri).toHaveBeenCalledWith(sampleRequest, urlToCheck)
-        }
-      )
+    it('should throw a GetDataRedirect if licence-fulfilment page is false on the status', async () => {
+      const status = {
+        [CONTACT.page]: false
+      }
+      const mockRequest = getRequestMock({ permission: {}, status })
+      await expect(() => getData(mockRequest)).rejects.toThrowRedirectTo(CONTACT.uri)
     })
-  })
 
-  describe('checkNavigation', () => {
     it('should throw a GetDataRedirect if licence-fulfilment page is false on the status', async () => {
       const status = {
         [LICENCE_FULFILMENT.page]: false
       }
-      const permission = { licenceLength: '12M', isRenewal: true, licensee: {} }
-      const mockRequest = generateRequestMock(permission, '', status)
-
+      const mockRequest = getRequestMock({ status })
       await expect(() => getData(mockRequest)).rejects.toThrowRedirectTo(LICENCE_FULFILMENT.uri)
     })
 
     it('should throw a GetDataRedirect if licence-confirmation page is false on the status', async () => {
       const status = {
         [LICENCE_CONFIRMATION_METHOD.page]: false
       }
-      const permission = { licenceLength: '12M', isRenewal: true, licensee: {} }
-      const mockRequest = generateRequestMock(permission, '', status)
-
+      const mockRequest = getRequestMock({ status })
       await expect(() => getData(mockRequest)).rejects.toThrowRedirectTo(LICENCE_CONFIRMATION_METHOD.uri)
     })
   })
```

Comment:
I've noticed that in contact-summary/route.js, there is the following:
```
  status.fromSummary = CONTACT_SUMMARY_SEEN
  await request.cache().helpers.status.setCurrentPermission(status)
```
As far as I can see, this isn't ever tested. Please can you add a couple of tests that fromSummary gets set and that request.cache().helpers.status.setCurrentPermission is called with the status object?